src/screens/SearchScreen.js

```diff
@@ -10,22 +10,25 @@ import ModalCloseButton from '../components/ModalCloseButton'
 import CitiesArr from '../components/CitiesArr'
 
 
-const SearchScreen = () => {
-  const [term, setTerm] = useState('')
+const SearchScreen = ({navigation}) => {
+  const [term, setTerm] = useState('ice cream')
   const [city, setCity] = useState('new york')
   const [modalVisible, setModalVisible] = useState(false)
   const [searchApi, results, errorMsg, contentReady] = useResults()
 
-  const filteredRes = FilterResByPrice(results)
+  const filteredRes = useMemo(() => 
+    FilterResByPrice(results), [results])
+
+  useEffect(() => {
+    if (results) {
+      searchApi(term, city)
+    }
+  }, [city])
 
   if (!results) {
     return null
   }
 
-  useEffect(() => {
-    searchApi(term, city)
-  }, [city])
-
```

Comment:
SearchScreen - one more bad mistake, was trying to check if the component was already fetched, with 'if' statement or set to 'null' if false, before useEffect hook. This can cause mistakes in hook logic. Put this check inside the useEffect hook. Now it works correctly. Also add useMemo hook to the filteredResusults, for better performance.